### PR TITLE
Refactor functions in wetdep.F90

### PR DESCRIPTION
**Pull request by tangshuaiqi**
this refactor includes three functions:

flux_precnum_vs_flux_prec_mpln
faer_resusp_vs_fprec_evap_mpln
fprecn_resusp_vs_fprec_evap_mpln

and the sub-functions called in them. All sub-functions are merged into the above three functions

### CODE REVIEW

File: components/eam/src/chemistry/aerosol/wetdep.F90 [hunk 1]
```diff
@@ -1149,267 +1149,146 @@ end subroutine wetdepa_v2
 
 
 !==============================================================================
+!     flux_precnum_vs_flux_prec_mp = precipitation number flux at the cloud base [drops/m^2/s]
+!     Options of assuming log-normal or marshall-palmer raindrop size distribution
+!
       function flux_precnum_vs_flux_prec_mpln( flux_prec, jstrcnv )
       real(r8) :: flux_precnum_vs_flux_prec_mpln
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/aerosol/wetdep.F90]: Do we know the units of these fluxes?

**Review by singhbalwinder — APPROVED** [review]
This looks great! You nearly got rid of about 125 lines of repeated code!! I have just one minor comment about the units.

File: components/eam/src/chemistry/aerosol/wetdep.F90 (hunk 1 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/aerosol/wetdep.F90]: pushed a new commit with units added and single-letter variables re-defined.

**Review by jeff-cohere — APPROVED** [review]

**Comment by singhbalwinder** [comment]
All tests passed:
```
$ ./cs.status.comp_tem1_wet_func_mam4_org_v2_baselines -s
comp_tem1_wet_func_mam4_org_v2_baselines: 5 tests
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
```
Merging to refactor-maint-2.0

**Pull request merged**